client/common/directives/ModelFieldReference/ModelFieldReference.js

```diff
@@ -140,7 +158,7 @@ angular.module('dashboard.directives.ModelFieldReference', [
         if (scope.options.api) apiPath = replaceSessionVariables(scope.options.api);
         GeneralModelService.list(apiPath, params, {preventCancel: true}).then(function(response) {
           if (!response) return; //in case http request was cancelled by newer request
-          scope.list = response;
+          scope.list = scope.selected.items ? removeSelectedFromList(scope.selected.items, response, scope.options.key) : response;
```

Comment:
Do we need the branch? Can we just always do:
`scope.list = removeSelectedFromlist()`?